consumer.go

```diff
@@ -785,11 +785,20 @@ func (c *Consumer) isPotentialExtraTopic(topic string) bool {
 	return false
 }
 
-func (c *Consumer) refreshMetadata() error {
-	err := c.client.RefreshMetadata()
-	if err == sarama.ErrTopicAuthorizationFailed {
-		// maybe we didn't have authorization to describe all topics
+func (c *Consumer) refreshMetadata() (err error) {
+	if c.client.config.Metadata.Full {
+		err = c.client.RefreshMetadata()
+	} else {
+		var topics []string
+		if topics, err = c.client.Topics(); err == nil && len(topics) != 0 {
+			err = c.client.RefreshMetadata(topics...)
```

Comment:
can we put len(topics) > 0  instead of len(topics) != 0 assuming we might get negative . But i don't think that will happen. just to be safe . you might ignore this if we know we won't get negative numbers .

Reply:
`len(slice)` can never be negative and `!=` is a cheaper operation than `>` - micro-optimisation